components/views/admin/dashboard.tsx

```diff
@@ -1,29 +1,83 @@
-import { GrowthChart } from "@/components/shared/GrowthChart";
-import { MentorsChart } from "@/components/shared/MentorsChart";
-import { StatsCards } from "@/components/shared/StatCard";
-import { UsersTable } from "@/components/shared/UserTable";
+"use client"
+
+import { GrowthChart } from "@/components/shared/admin/dashboard/GrowthChart";
+import { MentorsChart } from "@/components/shared/admin/dashboard/MentorsChart";
+import { StatsCards } from "@/components/shared/admin/dashboard/StatCard";
+import { UsersTable } from "@/components/shared/admin/dashboard/UserTable";
+import { endPoints } from "@/data/end-points";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+
+const BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL;
 
 export default function AdminView() {
+  const router = useRouter();
+
+  useEffect(() => {
+    const checkAccount = async () => {
+      try {
+        const user = localStorage.getItem("user");
+        const token = localStorage.getItem("token");
+
+        if (user && token) {
+          const userObj = JSON.parse(user);
+          const endPoint = `${BASE_URL}/${endPoints.adminAccount}/${userObj.accounts[0].id}`;
+          const response = await fetch(endPoint, {
+            method: "GET",
+            headers: {
+              "Content-Type": "application/json",
+              Authorization: `Bearer ${JSON.parse(token)}`,
+            },
+          });
+          
+
+          if (!response.ok) {
+            console.error("Failed to fetch account info", response);
+            throw new Error("Fetch failed");
+          }
+
+          const accountInfo = await response.json();
+
+          if (!accountInfo) {
+            console.error("AccountInfo doesn't exist",);
+            throw new Error("Account info not found");
```

Comment:
🤔check how you manage  throw new Error("Account info not found"); 
on this approach the error pop up in to client.